Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -16191,6 +16175,16 @@ void SpeculativeJIT::compileIdentity(Node* node)
     }
 }
 
+void SpeculativeJIT::compileExtractFromTuple(Node* node)
+{
+    RELEASE_ASSERT(node->child1().useKind() == UntypedUse);
+
+    ASSERT(m_graph.m_tupleData.at(node->tupleIndex()).virtualRegister == node->virtualRegister());
+    VirtualRegister virtualRegister = node->virtualRegister();
+    GenerationInfo& info = generationInfoFromVirtualRegister(virtualRegister);
+    info.stealForNode(node);
```

Comment:
Maybe, we should have a bit better name for that, like, `initFromTupleElement`?
And don't we need to setup this GenerationInfo based on resultFlags of tuple element? (As the same way to FTL)
Maybe, the producer side already initialize this GenerationInfo with appropriate types, and since we are using the same VirtualRegister here, we do not need to initialize it much, but maybe we should assert here to ensure that the resulted GenerationInfo matches against the expected Tuple element's type.